arch/arm/include/cortex_m/cmse.h

```diff
@@ -214,6 +214,122 @@ int arm_cmse_addr_range_readwrite_ok(u32_t addr, u32_t size, int force_npriv);
 #define ARM_CMSE_OBJECT_UNPRIV_READWRITE_OK(p_obj) \
 	cmse_check_pointed_object(p_obj, CMSE_MPU_UNPRIV | CMSE_MPU_READWRITE)
 
+#if defined(CONFIG_ARM_SECURE_FIRMWARE)
+
+/**
+ * @brief Get the MPU (Non-Secure) region number of an address
+ *
+ * Get the MPU (Non-Secure) region that the address maps to.
+ * Return non-zero to indicate that a valid MPU region was retrieved,
+ * and store the MPU region information in the supplied location.
+ *
+ * Note:
+ * Obtained region is valid only if:
+ * - the function is called from Secure state
+ * - the MPU is implemented and enabled
+ * - the given address matches a single, enabled MPU region
+ *
+ * @param addr The address for which the MPU region is requested
+ * @param p_region Output pointer to the location to store the MPU region
+ *
+ * @return non-zero if @ref region contains a valid MPU region, otherwise 0.
```

Comment:
This is opposite to the Zephyr approach, where 0 return value 0 is a success.
What about returning region number (>= 0) on success and -ERRCODE on failure?

Reply:
We could probably do that, yes. We would need to define an "arbitrary" ERRCODE and use it throughout the use of this API.

Reply:
That has to be done in all similar functions, of course.

Reply:
yeah let's conform to the usual zephyr return values.

Reply:
Good, I am going to update the PR with these changes. I am going to use -EINVAL to indicate invalid region.